### ISSUE TITLE
[Bug]: disallowedFor for forbid-component-props doesn't work as expected

### ISSUE THREAD

**Issue by edwarddamato** [jsx-eslint/eslint-plugin-react#3592]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

Despite defining which component a property is disallowed for, I still get eslint errors when I use the property on other components.

index.js
```jsx
import React from "react";

const SomeComponent = (props) => <div>foo</div>;

function App() {
  return (
    <div className="App">
      <SomeComponent context="hi"></SomeComponent>
    </div>
  );
}

export default App;

```

eslint config
```json
{
    "extends": [
        "eslint:recommended",
        "plugin:react/recommended"
    ],
    "env": {
        "es6": true
    },
    "root": true,
    "parserOptions": {
        "ecmaVersion": 2018,
        "sourceType": "module"
    },
    "settings": {
        "react": {
            "version": "detect"
        }
    },
    "plugins": [
        "react"
    ],
    "rules": {
        "react/forbid-component-props": [
            2,
            {
                "forbid": [
                    {
                        "propName": "context",
                        "disallowedFor": ["MyComponent"]
                    }
                ]
            }
        ]
    }
}

```

eslint says that the property `context` is not allowed on this component despite it being a different component than `MyComponent`.

![image](https://github.com/jsx-eslint/eslint-plugin-react/assets/19622548/48d9606a-febd-41d7-83aa-aebf42382e50)

The lint error shows both when running the eslint command as well as in the editor.

### Expected Behavior

I would expect that eslint does not show a linting error when using a property that is disallowed on other components.

The code shown in the 'Description Overview' should not show any lint errors related to forbidden component props.

### eslint-plugin-react version

v7.32.2

### eslint version

v8.43.0

### node version

v20.0.0

**Comment by ljharb** [comment]
That's because the option isn't released yet: https://github.com/jsx-eslint/eslint-plugin-react/commit/4a92667e4b99221013e1d2aa61e97296895cecc1

Everywhere on github, you always need to look at the docs on the version tag for the release you're using, and not master/main.
